AsyncDisplayKit/Layout/ASLayoutable.mm

```diff
@@ -27,14 +52,24 @@ static inline mach_port_t ASLayoutableGetCurrentContextKey()
 
 void ASLayoutableSetCurrentContext(struct ASLayoutableContext context)
 {
-  layoutableContextMap[ASLayoutableGetCurrentContextKey()] = context;
+  mach_port_t key = ASLayoutableGetCurrentContextKey();
+  ASDN::StaticMutexLocker l(_layoutableContextLock);
+  layoutableContextMap[key] = context;
 }
 
 struct ASLayoutableContext ASLayoutableGetCurrentContext()
 {
-  return layoutableContextMap[ASLayoutableGetCurrentContextKey()];
+  mach_port_t key = ASLayoutableGetCurrentContextKey();
+  ASDN::StaticMutexLocker l(_layoutableContextLock);
+  auto it = layoutableContextMap.find(key);
```

Comment:
Can you comment this code? It is not clear to me why we return the second, but I am not very familiar with this pattern. I assume this is an iterator though. This might be the only use of the auto keyword in the CodeBase